tools/restatectl/src/commands/log/seal_extend_chain.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::num::{NonZeroU32, NonZeroU8};
+
+use anyhow::Context;
+use cling::prelude::*;
+use tonic::codec::CompressionEncoding;
+use tonic::transport::Channel;
+
+use restate_admin::cluster_controller::protobuf::cluster_ctrl_svc_client::ClusterCtrlSvcClient;
+use restate_admin::cluster_controller::protobuf::{
+    ListLogsRequest, ProviderKind as ProtoProviderKind, SealAndExtendChainRequest,
+};
+use restate_types::logs::metadata::{Logs, ProviderKind};
+use restate_types::logs::LogId;
+use restate_types::protobuf::common::Version;
+use restate_types::replicated_loglet::{NodeSet, ReplicatedLogletParams, ReplicationProperty};
+use restate_types::storage::StorageCodec;
+use restate_types::{GenerationalNodeId, PlainNodeId};
+
+use crate::app::ConnectionInfo;
+use crate::util::grpc_connect;
+
+#[derive(Run, Parser, Collect, Clone, Debug)]
+#[clap()]
+#[cling(run = "seal_and_extend_chain")]
+pub struct SealAndExtendChainOpts {
+    /// LogId/Partition to seal and extend
+    #[clap(long)]
+    log_id: u32,
+    /// Option segment index to seal, default to tail. If provided
+    /// it has to be the last segment in the chain
+    #[clap(long)]
+    segment_index: Option<u32>,
+    /// Min metadata version
+    #[clap(long, default_value = "1")]
+    min_version: NonZeroU32,
+    /// Provider kind. One of [local, memory, replicated]
+    #[clap(long, default_value = "replicated")]
+    provider: ProviderKind,
+
+    /// Replication property
+    #[clap(long, required_if_eq("provider", "replicated"))]
+    replication_factor: Option<NonZeroU8>,
+    /// A comma-separated list of the nodes in the nodeset. e.g. N1,N2,N4 or 1,2,3
+    #[clap(long, required_if_eq("provider", "replicated"), value_delimiter=',', num_args = 1..)]
+    nodeset: Vec<PlainNodeId>,
```

Comment:
It'd be even nicer if it'd still work even with no arguments at all. In that case, it'd just seal and extend the chain with the same loglet configuration, but gets assigned a different loglet id (for all loglet types)

Reply:
When you say `extend  the chain with the same loglet configuration` does this include the `sequencer` in case of replicated loglet ? I don't get who this can be useful (except in the of offset range exhaustion)

Reply:
Yes. It's useful in manual testing